torch_xla/csrc/cross_replica_reduces.cpp

```diff
@@ -309,6 +309,28 @@ AllGatherResultCoalesced BuildAllGatherCoalesced(
   return {result, token_handler.GetNewToken(result[0])};
 }
 
+at::Tensor collective_broadcast(const at::Tensor& input, int64_t src,
+                                std::string) {
+  XLATensorPtr xinput = bridge::GetXlaTensor(input);
+  TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
+  at::Tensor mask;
+  const torch::lazy::BackendDevice& device = xinput->GetDevice();
+  if (device.ordinal() == src) {
+    mask = at::ones_like(input);
```

Comment:
Is there an equivalent to `torch.no_grad()` in C++?  That's the only difference I see between the original python version and this one

Reply:
Searched the doc and we can use the following scope for tensor operation without grad:
```
  {
    at::NoGradGuard no_grad;
    // tensor operations
   }
```

Reply:
Anyone knows why we set no grad here: https://github.com/pytorch/xla/blob/c0501f01514216b3b86f66cfe9b15dbe1d23a0d6/torch_xla/_internal/c10d_registration.py#L17?  @JackCaoG